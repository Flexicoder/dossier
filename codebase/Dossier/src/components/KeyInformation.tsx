import React, { Component } from 'react';
import { Text, TextInput, View, Platform } from 'react-native';
import DateTimePicker from '@react-native-community/datetimepicker';
import  styles  from '../assets/styles'

interface KeyInformationProps {
    name: string;
    preferredName?: string | null;
    dateOfBirth?: Date;
    address?: string | null;
    contactNumber?: string | null;
    email?: string | null;
    nhsNumber?: string | null;
}

export default class KeyInformation extends Component<KeyInformationProps> {
    constructor(props: KeyInformationProps) {
        super(props)
        this.state = { name: props.name,
                       preferredName: props.preferredName,
                       dateOfBirth: props.dateOfBirth || new Date(),
                       address: props.address,
                       contactNumber: props.contactNumber,
                       email: props.email,
                       nhsNumber: props.nhsNumber }
    }

    onChange = (event: any, selectedDate: Date | null | undefined) => {
        const currentDate = selectedDate || this.props.dateOfBirth;
        this.setState( { dateOfBirth: currentDate });
      };

    render() {
        return (
        <View >
                <View style={styles.line}>
                    <View style={styles.textBox}>
                        <TextInput
                            placeholder="Your Name"
                            onChangeText={text => this.setState({ name: text})}
                            defaultValue={this.props.name} />
                    </View>
                </View>
                <View style={styles.line}>
                    <View style={styles.textBox}>
                        <TextInput
                            placeholder="Preferred Name"
                            onChangeText={text => this.setState( { preferredName: text })}
                            defaultValue={this.props.preferredName || ""}
                            />
                    </View>
                </View>

                <Text>Date Of Birth</Text>
                <DateTimePicker
                    value={this.state.dateOfBirth}
                    onChange={this.onChange}
                    />

                <View style={styles.line}>
                    <View style={styles.textBox}>
                        <TextInput
                            placeholder="Address"
                            onChangeText={text => this.setState( { address: text })}
                            defaultValue={this.props.address || ""}
                            />
                    </View>
                </View>
                <View style={styles.line}>
                    <View style={styles.textBox}>
                        <TextInput
                            placeholder="Contact Number"
                            onChangeText={text => this.setState( { contactNumber: text })}
                            defaultValue={this.props.contactNumber || ""}
                            />
                    </View>
                </View>
                <View style={styles.line}>
                    <View style={styles.textBox}>
                        <TextInput
                            placeholder="Email"
                            onChangeText={text => this.setState( { email: text })}
                            defaultValue={this.props.email || ""}
                            />
                    </View>
                </View>
                <View style={styles.line}>
                    <View style={styles.textBox}>
                        <TextInput
                            placeholder="NHS Number"
                            onChangeText={text => this.setState( { nhsNumber: text })}
                            defaultValue={this.props.nhsNumber || ""}
                            />
                    </View>
                </View>
            </View>
        );
    }
}