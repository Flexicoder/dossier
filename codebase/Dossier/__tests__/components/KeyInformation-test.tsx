
import 'react-native';
import React from 'react';
import KeyInformation from '../../src/components/KeyInformation';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe("Key Information", () => {
    it('renders correctly', () => {
        const keyInformation = renderer.create(<KeyInformation name="" dateOfBirth={ new Date(1588595312810)}/>);
        expect(keyInformation.toJSON()).toMatchSnapshot();
      });
})


describe("Key Information", () => {
    it('contains supplied data', () => {
        const keyInformation = renderer.create(<KeyInformation name="Bob" dateOfBirth={ new Date(1588595312810)} />);
        const stringVersion = JSON.stringify(keyInformation.toJSON())
        expect(stringVersion).toMatch(/Bob/);
      });
})