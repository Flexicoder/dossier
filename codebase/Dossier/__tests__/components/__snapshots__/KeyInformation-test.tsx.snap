// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Key Information renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "margin": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderColor": "lightgrey",
          "borderWidth": 1,
          "color": "grey",
          "flexGrow": 1,
          "fontSize": 18,
          "margin": 5,
          "padding": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue=""
        onChangeText={[Function]}
        placeholder="Your Name"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "margin": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderColor": "lightgrey",
          "borderWidth": 1,
          "color": "grey",
          "flexGrow": 1,
          "fontSize": 18,
          "margin": 5,
          "padding": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue=""
        onChangeText={[Function]}
        placeholder="Preferred Name"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <Text>
    Date Of Birth
  </Text>
  <RNDateTimePicker
    date={1588595312810}
    mode="date"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "height": 216,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "margin": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderColor": "lightgrey",
          "borderWidth": 1,
          "color": "grey",
          "flexGrow": 1,
          "fontSize": 18,
          "margin": 5,
          "padding": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue=""
        onChangeText={[Function]}
        placeholder="Address"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "margin": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderColor": "lightgrey",
          "borderWidth": 1,
          "color": "grey",
          "flexGrow": 1,
          "fontSize": 18,
          "margin": 5,
          "padding": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue=""
        onChangeText={[Function]}
        placeholder="Contact Number"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "margin": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderColor": "lightgrey",
          "borderWidth": 1,
          "color": "grey",
          "flexGrow": 1,
          "fontSize": 18,
          "margin": 5,
          "padding": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue=""
        onChangeText={[Function]}
        placeholder="Email"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "margin": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderColor": "lightgrey",
          "borderWidth": 1,
          "color": "grey",
          "flexGrow": 1,
          "fontSize": 18,
          "margin": 5,
          "padding": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue=""
        onChangeText={[Function]}
        placeholder="NHS Number"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
</View>
`;
